src/app/OperationalDeviceProxy.cpp

```diff
@@ -55,6 +55,11 @@ void OperationalDeviceProxy::MoveToState(State aTargetState)
                       ChipLogValueX64(mPeerId.GetCompressedFabricId()), ChipLogValueX64(mPeerId.GetNodeId()), to_underlying(mState),
                       to_underlying(aTargetState));
         mState = aTargetState;
+
+        if (aTargetState != State::Connecting)
+        {
+            CleanupCASEClient();
+        }
```

Comment:
It doesn't feel right to put the statement here. This should be in the exit action of Connecting state. But I can't find a better place for this.

May be we should use a state machine class here, I saw @msandstedt have written a state machine class, but no one is using it, we can complete it and use it here in follow ups.

Reply:
Yes, I agree, a better-defined state machine would be good here and in various other places (e.g. in a lot of the IM classes...)